addons/mass_mailing/models/mailing.py

```diff
@@ -269,6 +306,11 @@ def create(self, values):
     def write(self, values):
         if values.get('body_html'):
             values['body_html'] = self._convert_inline_images_to_urls(values['body_html'])
+        if values.get('calendar_date'):
+            if any(mailing.schedule_type != 'scheduled' for mailing in self):
```

Comment:
Couldn't we check that mailing is "in_queue" (and draft probably) state(s) instead ? By the way, how are displayed in calendar view unscheduled mailings (just put in queue) ?

For example we have a "next_departure" field. Maybe its logic could be integrated / copied into the schedule flow, order to always have a date: sent date, or scheduled, or "next departure". Then when moving mailing in calendar view, we force schedule_date, only if mailing is in draft or in queue.

That way we keep a state-based model ... not sure, just proposing.